module/os/linux/zfs/zpl_file_range.c

```diff
@@ -147,18 +147,36 @@ zpl_remap_file_range(struct file *src_file, loff_t src_off,
 	 * doesn't do that, so we just turn the flag off.
 	 */
 	flags &= ~REMAP_FILE_CAN_SHORTEN;
+	if (flags & REMAP_FILE_DEDUP) {
+		/* Zero length means to clone everything to the end of the file
+		 */
+		if (len == 0)
+			len = i_size_read(file_inode(src_file)) - src_off;
```

Comment:
```suggestion
	/* Zero length means to clone everything to the end of the file */
	if (len == 0)
		len = i_size_read(file_inode(src_file)) - src_off;

	if (flags & REMAP_FILE_DEDUP) {
```

this must stay on top of the `REMAP_FILE_DEDUP` condition. Because if the `REMAP_FILE_DEDUP`-flag isn't set we still need to calculate the `len` to the end of the file.

Reply:
Yep.  I already fixed that but push the update.